File: 1251-longest-chunked-palindrome-decomposition/longest-chunked-palindrome-decomposition.java

class Solution {
    public int longestDecomposition(String text) {
        int n = text.length(), res = 0;
        String left = "", right = "";
        for (int i = 0; i < n; ++i) {
            left += text.charAt(i);
            right = text.charAt(n - i - 1) + right;
            if (left.equals(right)) {
                res++;
                left = "";
                right = "";
            }
        }
        return res;
    }
}

File: 1251-longest-chunked-palindrome-decomposition/longest-chunked-palindrome-decomposition.py
class Solution:
    def longestDecomposition(self, text: str) -> int:
        n = len(text)
        left = 0
        right = n - 1
        chunks = 0
        while left <= right:
            candidate_length = 1
            while (
                left + candidate_length - 1 < right - candidate_length + 1
                and text[left:left + candidate_length]
                != text[right - candidate_length + 1:right + 1]
            ):
                candidate_length += 1
            if (
                left + candidate_length -1 < right - candidate_length + 1
                and text[left:left + candidate_length]
                == text[right - candidate_length +1:right + 1]
            ):
                chunks += 2
                left += candidate_length
                right -= candidate_length
            else:
                chunks += 1
                break
        return chunks